src/java/picard/analysis/CollectNameSortedMetrics.java

```diff
@@ -0,0 +1,275 @@
+                    useRead2 = false;
+                } else {
+                    final int lowBQcount = getLowQualityBaseCount(pairToAnalyze.read2);
+                    basesExcludedByBaseq += lowBQcount;
+                    usableBaseCount -= lowBQcount;
+                }
+
+                // subtract out bad bases from overlaps between the reads
+                if (useRead1 && useRead2) {
+                    final int overlapCount = getOverlappingBaseCount(pairToAnalyze.read1, pairToAnalyze.read2);
+                    basesExcludedByOverlap += overlapCount;
+                    usableBaseCount -= overlapCount;
+                }
+
+                metrics.TOTAL_USABLE_BASES += usableBaseCount;
+
+                // TODO -- keep track of high quality soft-clipped bases so they don't count for mean coverage?
+            }
+
+            // record progress
+            progress.record(pairToAnalyze.read1);
+            if (isProperPair) {
+                progress.record(pairToAnalyze.read2);
+            }
+
+            // iterate
+            pairToAnalyze = getNextReadPair(iterator);
+        }
+
+        // finalize metrics
+        metrics.GENOME_TERRITORY = reader.getFileHeader().getSequenceDictionary().getReferenceLength();
+        metrics.MEAN_COVERAGE = metrics.TOTAL_USABLE_BASES / (double)metrics.GENOME_TERRITORY;
+        metrics.PCT_EXC_DUPE = basesExcludedByDupes / (double)metrics.TOTAL_BASES;
+        metrics.PCT_EXC_MAPQ = basesExcludedByMapq / (double)metrics.TOTAL_BASES;
+        metrics.PCT_EXC_UNPAIRED = basesExcludedByPairing / (double)metrics.TOTAL_BASES;
+        metrics.PCT_EXC_BASEQ = basesExcludedByBaseq / (double)metrics.TOTAL_BASES;
+        metrics.PCT_EXC_OVERLAP = basesExcludedByOverlap / (double)metrics.TOTAL_BASES;
+        final double totalExcludedBases = basesExcludedByDupes + basesExcludedByMapq + basesExcludedByPairing + basesExcludedByBaseq + basesExcludedByOverlap;
+        metrics.PCT_EXC_TOTAL = totalExcludedBases / metrics.TOTAL_BASES;
+
+        final MetricsFile<SeqMetrics, Integer> out = getMetricsFile();
+        out.addMetric(metrics);
+        out.write(OUTPUT);
+
+        return 0;
+    }
+
+    protected SeqMetrics generateMetrics() {
+        return new SeqMetrics();
+    }
+
+    /**
+     * Get the count of low quality bases in the given read
+     *
+     * @param read  the read
+     * @return non-negative int
+     */
+    private int getLowQualityBaseCount(final SAMRecord read) {
+        int count = 0;
+        for (final byte qual : read.getBaseQualities()) {
+            if (qual < MINIMUM_BASE_QUALITY)
+                count++;
+        }
+        return count;
+    }
+
+    /**
+     * Get the count of overlapping bases for the given reads
+     *
+     * @param read1  the 1st read
+     * @param read2  the 2nd read
+     * @return non-negative int
+     */
+    private int getOverlappingBaseCount(final SAMRecord read1, final SAMRecord read2) {
```

Comment:
I think technically this method should examine the bases in the overlap and ensure that in both R1 and R2 the bases in question are above the low quality base threshold.  Otherwise you'll take a double-hit on those bases.